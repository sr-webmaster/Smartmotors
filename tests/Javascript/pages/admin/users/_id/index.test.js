import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import createStore from 'tests/createStore'
import Page from '~/pages/admin/users/_id/index.vue'
import { FIXTURE_USER } from 'tests/__data__/user'
import VueRouter from 'vue-router'

describe('Edit User Page', () => {
  let mock, localVue, router

  router = new VueRouter()

  beforeEach(() => {
    const v = createLocalVue()
    localVue = v.localVue
    localVue.use(VueRouter)
    mock = v.mock

    mock
      .onGet('api/users/1').reply(200, { data: FIXTURE_USER })
      .onGet('api/currentUser').reply(200, FIXTURE_USER)
      .onAny().reply(config => {
        // Fault-tolerantly ignore api/companies
        if (config.url.indexOf('s/') >= -1) {
          return [200, { data: [] }]
        }
        console.warn('No mock match for ' + config.url, config)
        return [200, { data: [] }]
      })
  })
  afterEach(() => {
    mock.restore()
  })

  describe('Snapshots', () => {
    test('mocking data', (done) => {
      const store = createStore()

      const wrapper = mount(Page, {
        localVue,
        router,
        store
      })

      Page.beforeRouteEnterOrUpdate(wrapper.vm, { params: { id: 1 } }, null, async () => {
        await wrapper.vm.$nextTick()
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
  })
})
