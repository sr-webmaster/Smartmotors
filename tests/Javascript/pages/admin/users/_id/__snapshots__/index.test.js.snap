// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit User Page Snapshots mocking data 1`] = `
<div
  class="container fill-height"
>
  <div
    class="layout align-start row wrap"
  >
    <div
      class="flex xs12 sm12 md4 lg3 xl3 mb-4"
    >
      <div
        class="fresh-profile-card v-card v-sheet theme--light"
      >
        <div
          class="layout align-start row wrap"
        >
          <div
            class="flex xs12"
          >
            <div
              class="layout xs12 sm12 md12 lg12 row px-4 py-1 align-center"
            >
              <div
                class="flex pt-2"
              >
                <div
                  class="card-heading body-2 grey--text font-weight-bold"
                >
                  
      YOUR PROFILE
    
                </div>
              </div>
               
              <div
                class="spacer"
              />
                
              <a
                class="v-btn v-btn--icon theme--light"
                href="admin/users/1/edit"
              >
                <div
                  class="v-btn__content"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon grey--text fa fa fa fa-cog theme--light"
                    style="font-size: 16px;"
                  />
                </div>
              </a>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light fresh-tipmenu secondary"
                  style="max-height: auto; min-width: 0px; max-width: 300px; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
                >
                   
                  <span
                    class="font-weight-bold text-uppercase small-caption white--text text-no-wrap"
                  >
                    
      Edit Your Profile
    
                  </span>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="flex xs12 sm6 md12"
          >
            <div
              class="v-responsive v-image"
              style="max-height: 320px;"
            >
              <div
                class="v-image__image v-image__image--preload v-image__image--contain"
                style="background-position: center center;"
              />
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="flex xs12 sm6 md12"
          >
            <div
              class="flex xs12 sm12 md12"
            >
              <div
                class="v-card__text"
              >
                <div
                  class="layout pt-0 pl-1 mt-0 justify-left"
                >
                  <div
                    class="subheading font-weight-bold secondary--text mb-2"
                  >
                    
              Demo Admin
            
                  </div>
                </div>
                 
                <div
                  class="layout pt-0 ma-0 justify-left"
                >
                  <span
                    class="v-chip ma-0 font-weight-bold v-chip--small theme--dark secondary"
                    depressed=""
                    tabindex="0"
                  >
                    <span
                      class="v-chip__content"
                    >
                      
              Platform Admin
            
                    </span>
                  </span>
                </div>
                 
                <div
                  class="flex my-1 column"
                />
                 
                <div
                  class="layout pt-2 pl-1 mt-0 justify-left"
                >
                  <div
                    class="profile-position greyish-brown--text"
                  >
                    
              No Title @ Laravel
            
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex hidden-sm-only xs12"
            >
              <hr
                class="v-divider theme--light"
              />
            </div>
             
            <div
              class="flex xs12 sm12 md12"
            >
              <div
                class="v-list fresh-profile-card__contact-list theme--light"
                role="list"
              >
                <div
                  class="fresh-profile-card__contact-list-item"
                  role="listitem"
                >
                  <div
                    class="v-list__tile v-list__tile--avatar theme--light"
                  >
                    <div
                      class="v-list__tile__avatar fresh-profile-card__contact-list-item"
                    >
                      <div
                        class="v-avatar"
                        style="height: 40px; width: 40px;"
                      >
                        <div
                          class="v-avatar secondary"
                          style="height: 30px; width: 30px;"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon fa fa fa fa-envelope theme--dark"
                            style="font-size: 16px;"
                          />
                        </div>
                      </div>
                    </div>
                     
                    <div
                      class="v-list__tile__content"
                    >
                      <div
                        class="v-list__tile__title body-2"
                      >
                        
                demoAdmin@example.com
              
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="fresh-profile-card__contact-list-item"
                  role="listitem"
                >
                  <div
                    class="v-list__tile v-list__tile--avatar theme--light"
                  >
                    <div
                      class="v-list__tile__avatar fresh-profile-card__contact-list-item"
                    >
                      <div
                        class="v-avatar"
                        style="height: 40px; width: 40px;"
                      >
                        <div
                          class="v-avatar secondary"
                          style="height: 30px; width: 30px;"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon fa fa fa fa-phone theme--dark"
                            style="font-size: 16px;"
                          />
                        </div>
                      </div>
                    </div>
                     
                    <div
                      class="v-list__tile__content"
                    >
                      <div
                        class="v-list__tile__title body-2"
                      >
                        
                123-456-7890
              
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="fresh-profile-card__contact-list-item"
                  role="listitem"
                >
                  <div
                    class="v-list__tile v-list__tile--avatar theme--light"
                  >
                    <div
                      class="v-list__tile__avatar fresh-profile-card__contact-list-item"
                    >
                      <div
                        class="v-avatar"
                        style="height: 40px; width: 40px;"
                      >
                        <div
                          class="v-avatar secondary"
                          style="height: 30px; width: 30px;"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon fa fa fa fa-mobile-alt theme--dark"
                            style="font-size: 16px;"
                          />
                        </div>
                      </div>
                    </div>
                     
                    <div
                      class="v-list__tile__content"
                    >
                      <div
                        class="body-2"
                      >
                        
                321-123-1234
              
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="flex xs12 sm12 md8 lg9 xl9 pl-3"
    >
      <div
        class="layout row wrap"
      >
        <div
          class="flex xs12 mb-4 px-0"
        >
          <div
            class="v-card v-sheet theme--light"
            user="[object Object]"
          >
            <div
              class="layout xs12 sm12 md12 lg12 row px-4 py-1 align-center"
            >
              <div
                class="flex pt-2"
              >
                <div
                  class="card-heading body-2 grey--text font-weight-bold"
                >
                  
      YOUR COMPANY
    
                </div>
              </div>
               
              <div
                class="spacer"
              />
                
              <a
                class="v-btn v-btn--icon theme--light"
                href="admin/companies/1/edit"
              >
                <div
                  class="v-btn__content"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon grey--text fa fa fa fa-cog theme--light"
                    style="font-size: 16px;"
                  />
                </div>
              </a>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light fresh-tipmenu secondary"
                  style="max-height: auto; min-width: 0px; max-width: 300px; top: 12px; left: 12px; transform-origin: top left; z-index: 0; display: none;"
                >
                   
                  <span
                    class="font-weight-bold text-uppercase small-caption white--text text-no-wrap"
                  >
                    
      Edit Your Company
    
                  </span>
                </div>
              </div>
            </div>
             
            <hr
              class="v-divider theme--light"
            />
             
            <div
              class="container my-0 mx-2 px-2 py-2 justify-space-between"
            >
              <div
                class="layout ma-0 pa-0 justify-space-between row wrap"
              >
                <div
                  class="flex"
                >
                  <div
                    class="container grid-list-md fluid justify-start pa-0 ma-0"
                  >
                    <div
                      class="layout row justify-start align-center"
                    >
                      <div
                        class="v-avatar v-avatar--tile"
                        style="height: 48px; width: 48px;"
                      >
                        <div
                          class="v-responsive v-image"
                        >
                          <div
                            class="v-responsive__sizer"
                            style="padding-bottom: 100%;"
                          />
                          <div
                            class="v-image__image v-image__image--preload v-image__image--cover"
                            style="background-position: center center;"
                          />
                          <div
                            class="v-responsive__content"
                          />
                        </div>
                      </div>
                       
                      <div
                        class="flex grow"
                      >
                        <div
                          class="body-2 font-weight-bold secondary--text mb-1 mt-1"
                        >
                          
                Laravel
              
                        </div>
                         
                        <div>
                          <span
                            class="v-chip my-0 ml-0 mr-1 font-weight-bold text-uppercase v-chip--small theme--dark secondary"
                            depressed=""
                            tabindex="0"
                          >
                            <span
                              class="v-chip__content"
                            >
                              
    No Company Type
  
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex pr-1"
                >
                  <div
                    class="container grid-list-md fluid pa-0 ma-0"
                  >
                    <div
                      class="layout row justify-start"
                    >
                      <div
                        class="v-avatar"
                        style="height: 56px; width: 56px;"
                      >
                        <img
                          src="https://cdn.vuetifyjs.com/images/lists/2.jpg"
                        />
                      </div>
                       
                      <div
                        class="flex"
                      >
                        <div
                          class="body-2 font-weight-bold secondary--text mb-1 mt-1"
                        >
                          
                
              
                        </div>
                         
                        <span
                          class="v-chip ma-0 font-weight-bold text-uppercase small-caption v-chip--small theme--dark secondary"
                          depressed=""
                          tabindex="0"
                        >
                          <span
                            class="v-chip__content"
                          >
                            
                
              
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex shrink align-center mt-0 pr-2"
                >
                  <div
                    class="container grid-list-md fluid pa-0 ma-0 mr-4"
                  >
                    <div
                      class="layout row justify-start align-center"
                    >
                      <div
                        class="flex shrink"
                      >
                        <div
                          class="v-avatar secondary"
                          style="height: 50px; width: 50px;"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon white--text far fa-id-badge theme--light"
                            style="font-size: 28px;"
                          />
                        </div>
                      </div>
                       
                      <div
                        class="flex grow"
                      >
                        <div
                          class="headline font-weight-bold"
                        >
                          
                0
              
                        </div>
                         
                        <div
                          class="text-uppercase small-caption grey--text font-weight-bold"
                        >
                          
                Total Members
              
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex xs12 mb-4 pa-0"
        >
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
